src/main/java/fr/inria/jtravis/helpers/RepositoryHelper.java

```diff
@@ -18,16 +21,40 @@ public RepositoryHelper(JTravis jTravis) {
     }
 
     public Optional<Repository> fromSlug(String slug) {
+        Optional<Repository> repo = null;
+
         Optional<String> optionalS = this.getEncodedSlug(slug);
         if (optionalS.isPresent()) {
-            return getEntityFromUri(Repository.class, Arrays.asList(
+            repo = getEntityFromUri(Repository.class, Arrays.asList(
                     TravisConstants.REPO_ENDPOINT,
                     optionalS.get()), null);
+            if (!repo.isPresent()) {
```

Comment:
I'm not a huge fan of this check: IMHO we should only try to search using GH API when we previously got a 404 response: in other cases, it's using GH API for nothing...

Reply:
Note that it means, we do not necessarily use the getEntityFromUri or we might have to tweak it => maybe it does not return an optional but an object containing also the name of the obtained error in case the value is not present.